Flickr/Sources/FlickrAppVCs/PhotoViewerVC.swift

```diff
@@ -0,0 +1,161 @@
+        NSLayoutConstraint.activate(iconConstraints)
+        view.addSubview(stackView)
+
+        guard let imageView = imageView else {
+            return []
+        }
+        return [
+            stackView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 15.0),
+            stackView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -15.0),
+            stackView.topAnchor.constraint(equalTo: imageView.bottomAnchor, constant: 20.0)
+        ]
+    }
+
+    private func returnLabel(stackView: UIStackView) -> [NSLayoutConstraint] {
+        let label = UILabel()
+        label.configureView { label in
+            label.text = imageTitle
+            label.font = label.font.withSize(20)
+            label.clipsToBounds = false
+        }
+        stackView.addArrangedSubview(label)
+
+        return [
+            label.topAnchor.constraint(equalTo: stackView.topAnchor),
+            label.trailingAnchor.constraint(equalTo: stackView.trailingAnchor, constant: -60)
+        ]
+    }
+
+    private func returnIcon(stackView: UIStackView) -> [NSLayoutConstraint] {
+        let enabledButton = UIButton()
+        enabledButton.configureView { button in
+            button.imageView?.contentMode = .scaleAspectFit
+        }
+        let filledIcon = filledHeartIcon
+        enabledButton.isHidden = true
+        enabledButton.setImage(filledIcon, for: .normal)
+        enabledButton.addTarget(self, action: #selector(clickedFavorite), for: .touchUpInside)
+        favoriteEnabledButton = enabledButton
+
+        let disabledButton = UIButton()
+        disabledButton.configureView { button in
+            button.imageView?.contentMode = .scaleAspectFit
+        }
+        let outlinedIcon = outlinedHeartIcon
+        disabledButton.setImage(outlinedIcon, for: .normal)
+        disabledButton.isHidden = false
+        disabledButton.addTarget(self, action: #selector(clickedFavorite), for: .touchUpInside)
+        favoriteDisabledButton = disabledButton
+
+        stackView.addArrangedSubview(enabledButton)
+        stackView.addArrangedSubview(disabledButton)
+
+        return [
+            enabledButton.topAnchor.constraint(equalTo: stackView.topAnchor),
+            disabledButton.topAnchor.constraint(equalTo: stackView.topAnchor)
+        ]
+    }
+
+    private func setupImageView(stackView _: UIStackView?) -> [NSLayoutConstraint] {
+        let imageView = UIImageView()
+        imageView.configureView { imageView in
+            imageView.clipsToBounds = true
+            imageView.contentMode = .scaleAspectFit
+            imageView.backgroundColor = imageViewBackgroundColor
+        }
+        self.imageView = imageView
+        Nuke.loadImage(with: url, into: imageView)
+        view.addSubview(imageView)
+
+        return [
+            imageView.topAnchor.constraint(equalTo: view.topAnchor),
+            imageView.heightAnchor.constraint(equalToConstant: view.frame.height / 2),
```

Comment:
`view.frame.height / 2` is this the best constraint to be set to get the closest to the original image dimensions?

Reply:
Each image in the results of the search are of different dimension. So, I've set a imageView height, and the image is being displayed in it.